cmd/daprovider/daprovider.go

```diff
@@ -0,0 +1,216 @@
+package main
```

Comment:
cmd/ should have only small things that are only relevant to the binaries.
so this file makes sense here, but the "das" and "dasserver" subdirs should be moved somewhere else.

I think I do like the fact you removed it from "arbstate".. maybe under "/daprovider" or somewhere else?